cognite/client/utils/_identifier.py

```diff
@@ -162,5 +185,13 @@ def load(
         return cls(identifiers=[Identifier(val) for val in all_identifiers], is_singleton=is_singleton)
 
 
-class SingletonIdentifierSequence(IdentifierSequence):
+class DataModelIdentifierSequence(IdentifierSequenceCore[DataModelIdentifier]):
```

Comment:
A DataModel is a specific resource type in the Data Modeling API.
```suggestion
class DataModelingIdentifierSequence(IdentifierSequenceCore[DataModelIdentifier]):
```